tsconfig.base.json

```diff
@@ -8,9 +8,10 @@
     "moduleResolution": "node",
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
-    "target": "es2015",
+    "allowUnreachableCode": false,
+    "target": "es2017",
```

Comment:
why is this needed? Check up on consequences.

Reply:
The way I see it, this change is sane enough. 

This has no consequences for the angular library, as the tsconfig for that sets the target to `es2017`. 
So instead, we could do as suggested here and set the base config to `es2017`, and remove it from the angular specific config. 

Furthermore, the cookbook has a browserslistrc that angular uses to determine the needed JS output. Running `npx browserslist`, I see the lowest supported versions as:

```
chrome 90
edge 89
firefox 78
ios_saf 13.0-13.1
safari 13
```
Which is all well within [support for es2017](https://caniuse.com/?search=es2017).

Reply:
We should consider to bump the target of tsconfig.lib.json (published angular lib) to es2017 since we are already 'promising' this and more by setting the 'lib' to es2018. To my knowledge we are not supplying polyfills to close that gap.